src/mbgl/text/placement.hpp

```diff
@@ -87,11 +87,18 @@ class CollisionGroups {
 
 class BucketPlacementParameters {
 public:
+    friend bool operator<(const BucketPlacementParameters& lhs, const BucketPlacementParameters& rhs) {
+        return lhs.sortKey < rhs.sortKey;
+    }
+    SymbolBucket& bucket;
```

Comment:
`BucketPlacementParameters` shall not know about `SymbolBucket`

Reply:
Would it be ok for `BucketPlacementParameters` to have a reference to `Bucket`?

Reply:
Could we instead just pre-sort `LayerPlacementData` in [RenderSymbolLayer::prepare()](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/layers/render_symbol_layer.cpp#L571) accordingly to the the symbol bucket sort key?

Reply:
Yep, changed to your suggestion. This means that there are now possibly multiple entries into LayerPlacementData per tile which we have to account for elsewhere